server/src/main/java/org/elasticsearch/cluster/InternalClusterInfoService.java

```diff
@@ -223,11 +219,7 @@ public void run() {
                             if (logger.isTraceEnabled()) {
                                 logger.trace("Scheduling next run for updating cluster info in: {}", updateFrequency.toString());
                             }
-                            try {
-                                threadPool.schedule(updateFrequency, executorName(), this);
-                            } catch (EsRejectedExecutionException ex) {
-                                logger.debug("Reschedule cluster info service was rejected", ex);
-                            }
+                            threadPool.scheduleUnlessShuttingDown(updateFrequency, executorName(), this);
```

Comment:
This is also altering behavior. Previously the `EsRejectedExecutionException` was always swallowed but now it is thrown if the thread pool is not yet shutdown. This might actually be ok though but I think somebody from the distributed team should have a closer look at this.

Reply:
Same comment as in https://github.com/elastic/elasticsearch/pull/37709#discussion_r250546896, given that `executorName()` is  a ThreadPool of scaling type we should be safe.